api-server/src/server/boot/challenge.js

```diff
@@ -370,69 +366,64 @@ function projectCompleted(req, res, next) {
     }
   }
 
-  return user
-    .getCompletedChallenges$()
-    .flatMap(() => {
-      const { alreadyCompleted, updateData } = buildUserUpdate(
-        user,
-        completedChallenge.id,
-        completedChallenge
-      );
+  try {
+    // This is an ugly hack to update `user.completedChallenges`
+    await user.getCompletedChallenges$().toPromise();
+  } catch (e) {
+    return next(e);
+  }
```

Comment:
Same point as above - is this necessary?

Reply:
Yes

Reply:
:thinking: